plugin/src/main/java/org/opensearch/ml/utils/error/ErrorMessage.java

```diff
@@ -50,14 +51,36 @@ protected String fetchReason() {
     }
 
     protected String fetchDetails() {
-        // Some exception prints internal information (full class name) which is security concern
-        return emptyStringIfNull(exception.getLocalizedMessage());
+        String msg;
+        // Prevent the method from exposing internal information such as internal ip address etc. that is a security concern.
+        if (hasInternalInformation(exception)) {
+            msg = decorateMessage(exception);
+        } else {
+            msg = exception.getLocalizedMessage();
+        }
+
+        return emptyStringIfNull(msg);
     }
 
     private String emptyStringIfNull(String str) {
         return str != null ? str : "";
     }
 
+    private Boolean hasInternalInformation(Throwable t) {
```

Comment:
see general comment I made earlier. This probably need to go under a flag of some sort in case a hosted service provider want to turn it on. An organization or individual running in a private environment might actually want this check.

Reply:
Thanks Samuel, I got your point. Usually we consider internal IPs as sensitive information for a service provider. Regarding an individual running, they can easily get the information in other ways, as it is a cluster issue like some nodes dropping.